sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/ShowPartitionsSuite.scala

```diff
@@ -81,6 +82,18 @@ trait ShowPartitionsSuiteBase extends command.ShowPartitionsSuiteBase {
         Row("p1=__HIVE_DEFAULT_PARTITION__"))
     }
   }
+
+  test("SPARK-33591: null as string partition literal value 'null' after setting legacy conf") {
```

Comment:
Could you move this test to `org.apache.spark.sql.execution.command.ShowPartitionsSuiteBase`. It should pass for DS v2 too.